docs: Update v3.7 `csrfPrevention` note to include `graphql-upload` (#6480)
apollographql/apollo-server: docs/source/security/cors.mdx

description: Control access to your server's resources
---

> > ⚠️ New in Apollo Server 3.7: we **highly recommend** that all users pass `csrfPrevention: true` to `new ApolloServer()` to protect your server from CSRF and XS-Search attacks. If you enable this feature, any clients that send operations via `GET` requests must include a particular header in that request. For more information, see [Preventing Cross-Site Request Forgery (CSRF)](#preventing-cross-site-request-forgery-csrf). Additionally, if you integrate `graphql-upload` into your server and enable `csrfPrevention: true`, your upload client must include a non-empty `Apollo-Require-Preflight` (or any other non-special-cased) header as part of the request. Without this header, the request won't be preflighted and Apollo Server will reject it with a helpful error message.

Reviewer: Hmm I was imagining kind of the opposite here. Less "here's something you need to do to make the flag work" and more "here is the set of people for whom we are going past 'highly recommend' to, um, whatever is stronger than 'highly recommend'".

Reviewer: (The previous sentence does already talk about "include a particular header" so we might just extend that to talk about uploads in addition to GETs)

description: Control access to your server's resources
---

> > ⚠️ New in Apollo Server 3.7: we **highly recommend** that all users pass `csrfPrevention: true` to `new ApolloServer()` to protect your server from CSRF and XS-Search attacks. If you enable this feature, any clients that send operations via `GET` requests must include a non-empty `Apollo-Require-Preflight` (or any other non-special-cased) header in that request. If you integrate `graphql-upload` into your server, your upload client must also include this header. For more information, see [Preventing Cross-Site Request Forgery (CSRF)](#preventing-cross-site-request-forgery-csrf). Additionally, if you integrate `graphql-upload` into your server and enable `csrfPrevention: true`, your upload client must include a non-empty `Apollo-Require-Preflight` (or any other non-special-cased) header as part of the request. Without this header, the request won't be preflighted and Apollo Server will reject it with a helpful error message.

Reviewer: ```suggestion
> ⚠️ New in Apollo Server 3.7: we **highly recommend** that all users pass `csrfPrevention: true` to `new ApolloServer()` to protect your server from CSRF and XS-Search attacks. **This is especially important if you integrate `graphql-upload` into your server.** If you enable this feature, any clients that send operations via `GET` requests or multipart upload requests must include a non-empty `Apollo-Require-Preflight` (or any other non-special-cased) header in that request. For more information, see [Preventing Cross-Site Request Forgery (CSRF)](#preventing-cross-site-request-forgery-csrf).
```
